docs/Contribute_To_Frontend_Quick_Guide.md

```diff
@@ -0,0 +1,243 @@
+* To modify and test any Frontend changes to IML, it will be necessary to install a working version of IML.
+    * Create a **Vagrant** virtual cluster outined here: [Install IML on a Vagrant Virtual Cluster](Installing_IML_on_HPC_Storage_Sandbox.md).
+    * Edit the **Vagrantfile** to allow for a shared mount. Immediately after the block that states *Create an admin server for the cluster*, add the following line:
+
+        **config.vm.synced_folder ".", "/vagrant", type: "virtualbox"**
+    
+![shared_mount](md_Graphics/vagrant_shared_mount.png)
+
+This will mount the current local directory to **/vagrant** on the virtual machine.
+
+ * Ensure the ability to log in to the **adm** node as the root user:
+    * vagrant up adm
+    * vagrant ssh
+    * su -
+    * ls -al /vagrant
+    * The Vagrantfile should be in the directory listing 
+
+## On the Local machine, i.e., not the vagrant virtual machine.
+
+For the desciption that follows, it will be assumed that the Vagrant file and virtual machine information reside in the directory:
+``` 
+~/vagrant-projects/vhpc 
+```
+## Contributing to the IML Frontend
+### Clone the Frontend repo, aka, the [GUI](https://github.com/intel-hpdd/GUI) repo
+```
+cd ~/vagrant-projects/vhpc
+git clone git@github.com:intel-hpdd/GUI.git 
+```
+### Create a branch, always work on a branch.
+```
+cd GUI
+git checkout -b  my-fix
+```
+### Validate that the correct branch has been selected.
+```
+git branch
+```
+### Work on the branch.
+```
+Use VS Code and open ~/vagrant-projects/vhpc/GUI
+```
+### As an Example, make a small change to the **Configuration** dropdown menu.
+In VS Code, locate and edit the following file: **app-states.js**
+```
+GUI 
+  > source
+    > iml
+      > app
+        > app-states.js
+
+Search for "Servers"
+```
+On the line containing "Servers", change **Servers** to **My Servers**:
+
+Change this line:
+```
+<li><a tabindex="-1" id="server-conf-item" ui-sref="app.server({ resetState: true })">Servers</a></li>
+```
+To look like this line:
+```
+<li><a tabindex="-1" id="server-conf-item" ui-sref="app.server({ resetState: true })">My Servers</a></li>
+```
+Save the file: **app-states.js**
+
+## Install dependencies and build the code
+
+### Install yarn
+Before you start using Yarn, you'll first need to install it on your system. 
+* How to [Install yarn](https://yarnpkg.com/lang/en/docs/install/)
+
+
+If yarn is installed properly, Install the external library dependencies
+
+Type the command:
+```
+yarn install
+```
+Build the code and Pass all required tests listed in **pacakage.json**
+```
+yarn watch
+```
+
+The watch command will leave the code in a state that will "watch" for further edits. If desired, the "watch" may be stopped by hitting ctrl-c
+
+### In a seperate terminal, log into the **adm** node
+* vagrant ssh adm
+
+Log in as root
+* su -
+
+Go to the cloned **GUI** repo on the /vagrant mount point where the code edits were made.
```

Comment:
This is now out of date. The GUI is being packaged as a copr repo, so the user will need to either symlink to that repo or have an overridden nginx.conf that points back to their mounted dir.